### ISSUE TITLE
Bug related to the explanation of a classification

### ISSUE THREAD

**Issue by RuiJardim** [TeamHG-Memex/eli5#311]
There may be bugs related to the explanation of a binary classification for the GradientBoostingClassifier algorithm of scikit learn.
I say this because there are very obvious features that are almost 100% sure to help with the decision, taking into account the distributions of values of both classes (positive and negative), and which are incredibly contributing negatively to the decision.
Can you check?

**Comment by lopuhin** [comment]
Thanks for the report @RuiJardim , could you please provide some details? E.g. some code you run, what you got and what you expected? For example it's not obvious if you are talking about ``explain_prediction`` or ``explain_weights``.

**Comment by RuiJardim** [comment]
I'm referring to the 'explain_prediction' method for explaining a given binary classification. Some verified issues:
- Sometimes the sum of the weights (shown in summary) does not match the sum of the weights of the feature listing that is displayed.
- For a classification referring to the negative class the sum of the weights has the opposite sign. For example it says it's -3000, but you should say it's +3000.
- Another issue is that the probability of the presented classification is not proportional to the weights of the features. For example, 3 features contributed negatively in that the sum of their weights is 400, and 9 features contributed positively in that the sum of their weights is 1600. Therefore the probability of the classification should be: 1600 / (1600 + 400) = 0.80. However, the probability of the classification (presented in summary) is correct, the proportion is that it does not work which also indicates that there may be some bug related to the presentation of the weights of the features.
